### ISSUE TITLE
Not able to retrieve metadata with  SFDX: Retrieve source from Org

### ISSUE THREAD

**Issue by IVerkhosh** [forcedotcom/cli#962]
### Summary

Getting: ERROR running force:source:retrieve: Unexpected file found in package directory. I fixed one bu adding audience to metadataTypeInfos.json file in the node_modules\salesforce-alm\metadata. 
But now it complaining about *\force-app\main\default\customindex\Account.AccountNumber.indx-meta.xml.
Even if I delete customindex folder, it start complaining about objects.

### Steps To Reproduce:

1.  Right click on package.xml 
1.  Select Retrieve source from Org

### Expected result
retrieved data in the package. xml file

### Actual result
08:57:18.339 sfdx force:source:retrieve --manifest c:\GIT_REPOS\ENERGY_INSIGHT_SFDX\manifest\package.xml
ERROR running force:source:retrieve:  Unexpected file found in package directory: c:\GIT_REPOS\ENERGY_INSIGHT_SFDX\force-app\main\default\customindex\Account.AccountNumber.indx-meta.xml
08:58:21.859 sfdx force:source:retrieve --manifest c:\GIT_REPOS\ENERGY_INSIGHT_SFDX\manifest\package.xml

### Additional information
@salesforce/sfdx-trust 3.6.0 (core)
alias 1.1.8 (core)
auth 1.5.1 (core)
config 1.2.7 (core)
generator 1.1.5 (core)
salesforcedx 51.6.0 (core)
├─ data 0.4.3 (core)
├─ schema 1.0.4 (core)
├─ limits 1.0.4 (core)
├─ apex 0.1.4 (core)
├─ custom-metadata 1.0.11 (core)
├─ org 1.6.0 (core)
├─ user 1.2.1 (core)
├─ templates 51.3.0 (core)
├─ salesforce-alm 51.6.6 (core)
└─ @salesforce/sfdx-plugin-lwc-test 0.1.7 (core)
sfdx-cli 7.94.3 (core)
telemetry 1.1.1 (core)
VS Code: Version: 1.55.0 (system setup)
Commit: c185983a683d14c396952dd432459097bc7f757f
Date: 2021-03-30T16:01:55.261Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.18363
**OS and version**: Windows 10

**Comment by IVerkhosh** [comment]
The same "Unexpected file found in package directory:" happened when using sfdx force:source:retrieve -m

**Comment by IVerkhosh** [comment]
Trying to debug the issue:
C:\GIT_REPOS\ENERGY_INSIGHT_SFDX>sfdx force:source:retrieve -m CustomTab --json --loglevel=debug
{
  "status": 1,
  "name": "UnexpectedFileFound",
  "message": "Unexpected file found in package directory: C:\\GIT_REPOS\\ENERGY_INSIGHT_SFDX\\force-app\\main\\default\\customindex\\Account.AccountNumber.indx-meta.xml",
  "exitCode": 1,
  "commandName": "SourceRetrieveCommand",
  "stack": "UnexpectedFileFound: Unexpected file found in package directory: C:\\GIT_REPOS\\ENERGY_INSIGHT_SFDX\\force-app\\main\\default\\customindex\\Account.AccountNumber.indx-meta.xml\n    at Workspace.isValidSourcePath (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\salesforce-alm\\dist\\lib\\source\\workspace.js:166:35)\n    at Workspace.handleArtifact (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\salesforce-alm\\dist\\lib\\source\\workspace.js:149:18)\n    at async Workspace.walk (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\salesforce-alm\\dist\\lib\\source\\workspace.js:134:36)\n    at async Workspace.walk (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\salesforce-alm\\dist\\lib\\source\\workspace.js:136:17)\n    at async Workspace.walk (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\salesforce-alm\\dist\\lib\\source\\workspace.js:136:17)\n    at async Workspace.walk (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\salesforce-alm\\dist\\lib\\source\\workspace.js:136:17)\n    at async Workspace.walkDirectories (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\salesforce-alm\\dist\\lib\\source\\workspace.js:121:13)\n    at async Workspace.initializeStateFull (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\salesforce-alm\\dist\\lib\\source\\workspace.js:90:9)\n    at async Workspace.init (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\salesforce-alm\\dist\\lib\\source\\workspace.js:37:17)\n    at async Function.create (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\@salesforce\\kit\\lib\\creatable.js:31:9)\n    at async SourcePathStatusManager.init (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\salesforce-alm\\dist\\lib\\source\\sourcePathStatusManager.js:185:26)\n    at async Function.create (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\@salesforce\\kit\\lib\\creatable.js:31:9)\n    at async SourceWorkspaceAdapter.init (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\salesforce-alm\\dist\\lib\\source\\sourceWorkspaceAdapter.js:60:21)\n    at async Function.create (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\@salesforce\\kit\\lib\\creatable.js:31:9)\n    at async SourceRetrieve.retrieve (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\salesforce-alm\\dist\\lib\\source\\sourceRetrieve.js:47:20)\n    at async SourceRetrieveCommand.execute (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\salesforce-alm\\dist\\lib\\source\\sourceRetrieveCommand.js:32:16)\nOuter stack:\n    at Function.wrap (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\@salesforce\\core\\lib\\sfdxError.js:171:27)\n    at SourceRetrieveCommand.catch (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\salesforce-alm\\dist\\ToolbeltCommand.js:248:46)\n    at async SourceRetrieveCommand._run (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\@salesforce\\command\\lib\\sfdxCommand.js:85:13)\n    at async Config.runCommand (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\@oclif\\config\\lib\\config.js:173:24)\n    at async Main.run (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\@oclif\\command\\lib\\main.js:27:9)\n    at async Main._run (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\node_modules\\@oclif\\command\\lib\\command.js:43:20)\n    at async Object.run (C:\\Users\\igor\\AppData\\Local\\sfdx\\client\\7.94.3-a4e7c7955b\\dist\\cli.js:114:21)",
  "warnings": []

**Comment by peternhale** [comment]
@IVerkhosh would it be possible to share your sfdx-project.json file? Also, if you could run `sfdx update` to advance to the latest release.

**Comment by IVerkhosh** [comment]
I already on the latest release: sfdx-cli/7.94.3-a4e7c7955b win32-x64 node-v14.15.4
sfdx-project.json file:
{
  "packageDirectories": [
    {
      "path": "force-app",
      "default": true
    }
  ],
  "namespace": "",
  "sfdcLoginUrl": "https://login.salesforce.com",
  "sourceApiVersion": "51.0"
}

**Comment by peternhale** [comment]
@IVerkhosh thank you. Are you free to share the contents of the manifest package.xml file you are using?

**Comment by IVerkhosh** [comment]
Hi @peternhale, I think I figured it out, by retrieving a few components at a time.
All retrievals went well, until I try to retrieve the CustomIndex. At this point, with the sfdx force:source:retrieve -m "CustomIndex" --json --loglevel=debug, I received the     "errorMessage": "Need to specify full name, Current Name: needfuel__Linked_Agreement__c., Required Delimiter: .". To clear it, I greated a new package.xml, for the CustomIndex only.
This list all custom indexes for me, and the last two have incorrect format. After removing those two entries, I was able to retrieve the CustomIndex with simple retrieve source from org.
